test/jest/metrics-server.spec.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ * SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+ */
+import { describe, test } from "@jest/globals";
+import * as k8s from "@kubernetes/client-node";
+import fetch from 'node-fetch';
+
+const kc = new k8s.KubeConfig();
+kc.loadFromDefault();
+
+interface KubeConfigCerts {
+  caCert?: string;
+  cert?: string;
+  key?: string;
+}
+
+function getApiServerAddress(): string | undefined {
+    try {
+      const currentContext = kc.getCurrentContext();
+      const context = kc.getContextObject(currentContext);
+  
+      if (!context) {
+        console.error(`Context "${currentContext}" not found in kubeconfig`);
+        return undefined;
+      }
+  
+      const clusterName = context.cluster;
+      const cluster = kc.getCluster(clusterName);
+  
+      if (!cluster) {
+        console.error(`Cluster "${clusterName}" not found in kubeconfig`);
+        return undefined;
+      }
+  
+      return cluster.server;
+    } catch (err) {
+      console.error('Error getting API server address:', err);
+      return undefined;
+    }
+  }
+
+function getCertsFromKubeConfig(): KubeConfigCerts | undefined {
+    try {
+      const currentContext = kc.getCurrentContext();
+      const context = kc.getContextObject(currentContext);
+  
+      if (!context) {
+        console.error(`Context "${currentContext}" not found in kubeconfig`);
+        return undefined;
+      }
+  
+      const clusterName = context.cluster;
+      const cluster = kc.getCluster(clusterName)
+      const user = kc.getCurrentUser()
+  
+      if (!cluster) {
+        console.error(`Cluster "${clusterName}" not found in kubeconfig`);
+        return undefined;
+      }
+
+      const caCert = cluster?.caData || "";
+      const cert = user?.certData || ""
+      const key = user?.keyData || ""
+      
+      return { caCert, cert, key };
+  
+    } catch (err) {
+      console.error('Error getting certs from kubeconfig:', err);
+      return undefined;
+    }
+  }
+
+describe("Metrics Server", () => {
+    const apiServerAddress = getApiServerAddress();
+    const certs = getCertsFromKubeConfig() || {};
+    const metricsApi = "/apis/metrics.k8s.io/v1beta1";
+    const https = require('https');
+
+    if (!apiServerAddress) {
+        console.error("API server address not found.");
```

Comment:
I might be misunderstanding this, if the apiServerAddress or the certs are missing we should fail the tests right? if so i think it would make more sense to throw an error. It would "fail fast", preventing jest from silently skipping test logic and being a bit more loud in CI if there is a failure.

```suggestion
        if (!apiServerAddress) {
            throw new Error("API server address not found.");
        }
```

Reply:
Good callout! Updated to `throw` errors instead of logging them.